Website/DesktopModules/AuthenticationServices/DNN/Login.ascx

```diff
@@ -39,39 +40,41 @@
         </div>
     </div>
 </div>
-<script type="text/javascript">
-	/*globals jQuery, window, Sys */
-	(function ($, Sys) {
-		function setUpLogin() {
-			var actionLinks = $("a#dnn_ctr<%#ModuleId > Null.NullInteger ? ModuleId.ToString() : ""%>_Login_Login_DNN_cmdLogin");
-			actionLinks.click(function () {
-				if ($(this).hasClass("dnnDisabledAction")) {
-					return false;
-				}
+<dnn:DnnScriptBlock runat="server">
+    <script type="text/javascript">
+        /*globals jQuery, window, Sys */
+        (function ($, Sys) {
+            function setUpLogin() {
+                var actionLinks = $("a#dnn_ctr<%=ModuleId > Null.NullInteger ? ModuleId.ToString() : ""%>_Login_Login_DNN_cmdLogin");
+                actionLinks.click(function () {
+                    if ($(this).hasClass("dnnDisabledAction")) {
+                        return false;
+                    }
 
-				actionLinks.addClass("dnnDisabledAction");
-			});
-		}
+                    actionLinks.addClass("dnnDisabledAction");
+                });
+            }
 		
-		$(document).ready(function () {
-		    $('.dnnLoginService').on('keydown', function (e) {
-				if ($(e.target).is('input:text,input:password') && e.keyCode === 13) {
-					var $loginButton = $('#dnn_ctr<%#ModuleId > Null.NullInteger ? ModuleId.ToString() : ""%>_Login_Login_DNN_cmdLogin');
-					if ($loginButton.hasClass("dnnDisabledAction")) {
-						return false;
-					}
+            $(document).ready(function () {
+                $(document).on('keydown', '.dnnLoginService', function (e) {
+                    if ($(e.target).is('input:text,input:password') && e.keyCode === 13) {
+                        var $loginButton = $('#dnn_ctr<%=ModuleId > Null.NullInteger ? ModuleId.ToString() : ""%>_Login_Login_DNN_cmdLogin');
```

Comment:
And what is the significance of changing the data-binding syntax here?

Reply:
if using <%#..., then must call DataBind from codebind to let the databinding effective, using <%=... doesn't need that. if calling DataBind manually, it may cause issue by life cycle changed.